File: src/test/java/calculator/TextParser.java
package calculator;

import java.util.Arrays;
import java.util.List;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

public class TextParser {
    private static final Pattern pattern = Pattern.compile("//(.)\n(.*)");

    public static List<Number> parseToExtractNumbers(String text) {
        var delimiters = new Delimiters(List.of(',', ';'));
        var matcher = pattern.matcher(text);
        var number = text;
        if (matcher.find()) {
            delimiters.add(matcher.group(1).charAt(0));
            number = matcher.group(2);
        }

        return Arrays.stream(number.split(delimiters.toRegex()))
            .map(Number::new)
            .collect(Collectors.toUnmodifiableList());
    }
}

File: src/test/java/calculator/TextParser.kt
package calculator

import java.util.regex.Pattern

class TextParser {
    companion object {
        private val DEFAULT_DELIMITERS = arrayOf(",", ":")
        private val CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)")
    }

    fun parse(text: String): Numbers {
        val matcher = CUSTOM_DELIMITER_PATTERN.matcher(text)

        return if (matcher.find()) {
            val customDelimiter = matcher.group(1)
            Numbers.create(arrayOf(customDelimiter), matcher.group(2))
        } else {
            Numbers.create(DEFAULT_DELIMITERS, text)
        }
    }
}